event/consumer_test.go

```diff
@@ -2,138 +2,150 @@ package event_test
 
 import (
 	"context"
-	"errors"
-	"github.com/ONSdigital/dp-search-data-importer/config"
-	"sync"
 	"testing"
+	"time"
+
+	"github.com/ONSdigital/dp-search-data-importer/config"
+	"github.com/ONSdigital/dp-search-data-importer/event/eventtest"
+	"github.com/ONSdigital/dp-search-data-importer/models"
 
-	kafka "github.com/ONSdigital/dp-kafka/v2"
 	"github.com/ONSdigital/dp-kafka/v2/kafkatest"
 	"github.com/ONSdigital/dp-search-data-importer/event"
-	"github.com/ONSdigital/dp-search-data-importer/event/mock"
-	"github.com/ONSdigital/dp-search-data-importer/schema"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
-var testCtx = context.Background()
-
-var errHandler = errors.New("Handler Error")
-
-var testEvent = event.HelloCalled{
-	RecipientName: "World",
-}
-
-// kafkaStubConsumer mock which exposes Channels function returning empty channels
-// to be used on tests that are not supposed to receive any kafka message
-var kafkaStubConsumer = &kafkatest.IConsumerGroupMock{
-	ChannelsFunc: func() *kafka.ConsumerGroupChannels {
-		return &kafka.ConsumerGroupChannels{}
-	},
-}
+var (
+	testCtx = context.Background()
+
+	expectedEvent1 = models.SearchDataImportModel{
+		DataType:        "testDataType1",
+		JobID:           "",
+		SearchIndex:     "ONS",
+		CDID:            "",
+		DatasetID:       "",
+		Keywords:        []string{"testkeyword1", "testkeyword2"},
+		MetaDescription: "",
+		Summary:         "",
+		ReleaseDate:     "",
+		Title:           "testTitle1",
+		TraceID:         "",
+	}
+
+	expectedEvent2 = models.SearchDataImportModel{
+		DataType:        "testDataType2",
+		JobID:           "",
+		SearchIndex:     "ONS",
+		CDID:            "",
+		DatasetID:       "",
+		Keywords:        []string{"testkeyword21", "testkeyword22"},
+		MetaDescription: "",
+		Summary:         "",
+		ReleaseDate:     "",
+		Title:           "",
+		TraceID:         "",
+	}
+)
 
-// TODO: remove or replace hello called logic with app specific
-func TestConsume(t *testing.T) {
+func TestConsumeWithOneMessage(t *testing.T) {
 
-	Convey("Given kafka consumer and event handler mocks", t, func() {
+	Convey("Given a consumer with a mocked message producer with an expected message", t, func() {
 
-		cgChannels := &kafka.ConsumerGroupChannels{Upstream: make(chan kafka.Message, 2)}
-		mockConsumer := &kafkatest.IConsumerGroupMock{
-			ChannelsFunc: func() *kafka.ConsumerGroupChannels { return cgChannels },
+		messageConsumer := kafkatest.NewMessageConsumer(false)
+		eventHandler := eventtest.NewEventHandler()
+		cfg, err := config.Get()
+		if err != nil {
+			t.Fatalf("failed to retrieve configuration: %v", err)
 		}
+		So(err, ShouldBeNil)
```

Comment:
Due to the above line 59 is not needed.

Think of it like this, we need the `config.Config` struct and we are getting this by calling `config.Get`; this is inherently part of the setup for the test. If this fails (errors), then we cannot run our test including all assertions.

Therefore the assertion on line 59 will always pass or this assertion wont be reached.

Reply:
removed, this line was introduced by one of the previous comments.